contracts/IdoContract/src/IdoContract.cs

```diff
@@ -382,9 +384,18 @@ public static bool SwapToken(UInt160 user, UInt160 idoPairContractHash, BigInteg
             SafeTransfer(GetSpendAssetHash(), user, Runtime.ExecutingScriptHash, spendAssetAmount);
             SafeTransfer(GetSpendAssetHash(), Runtime.ExecutingScriptHash, idoPairContractHash, spendAssetAmount);
             BigInteger balanceAfter = GetBalanceOfToken(project.tokenHash, Runtime.ExecutingScriptHash);
-            //ExecutionEngine.Assert(balanceAfter - balanceBefore == amount, "amount not correct");
-            AddUserClaimAmount(idoPairContractHash, user, amount);
-            AddUserSwapAmount(idoPairContractHash, user, amount);
+
+            BigInteger recAmount = balanceAfter - balanceBefore;
+            ExecutionEngine.Assert(recAmount > 0, "amount not correct");
```

Comment:
>= ?

Reply:
swapped amount == 0?

Reply:
I think the amount of swap can't be == 0